tests/integration-service/integration-with-env.go

```diff
@@ -0,0 +1,172 @@
+						return fmt.Errorf("build pipelineRun %s/%s still contains the finalizer: %s", pipelineRun.GetNamespace(), pipelineRun.GetName(), pipelinerunFinalizerByIntegrationService)
+					}
+					return nil
+				}, timeout, interval).Should(Succeed(), "timeout when waiting for finalizer to be removed")
+			})
+
+			It("checks if CronJob spacerequest-cleaner exists", func() {
+				spaceRequestCleanerCronJob, err := utils.GetCronJobs()
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(strings.Contains(spaceRequestCleanerCronJob, "spacerequest-cleaner")).Should(BeTrue())
+			})
+
+			It("checks if all of the integrationPipelineRuns passed", Label("slow"), func() {
+				Expect(f.AsKubeDeveloper.IntegrationController.WaitForAllIntegrationPipelinesToBeFinished(testNamespace, applicationName, snapshot)).To(Succeed())
+			})
+
+			It("checks if space request is created in namespace", func() {
+				spaceRequestsList, err := f.AsKubeAdmin.GitOpsController.GetSpaceRequests(testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(strings.Contains(spaceRequestsList.Items[0].GetName(), "task-spacerequest-")).Should(BeTrue())
+			})
+
+			It("checks if the passed status of integration test is reported in the Snapshot", func() {
+				timeout = time.Second * 240
+				interval = time.Second * 5
+				Eventually(func() error {
+					snapshot, err = f.AsKubeAdmin.IntegrationController.GetSnapshot(snapshot.Name, "", "", testNamespace)
+					Expect(err).ShouldNot(HaveOccurred())
+
+					statusDetail, err := f.AsKubeDeveloper.IntegrationController.GetIntegrationTestStatusDetailFromSnapshot(snapshot, integrationTestScenario.Name)
+					Expect(err).ToNot(HaveOccurred())
+
+					if statusDetail.Status != intgteststat.IntegrationTestStatusTestPassed {
+						return fmt.Errorf("test status for scenario: %s, doesn't have expected value %s, within the snapshot: %s", integrationTestScenario.Name, intgteststat.IntegrationTestStatusTestPassed, snapshot.Name)
+					}
+					return nil
+				}, timeout, interval).Should(Succeed())
+			})
+
+			It("checks if the finalizer was removed from all of the related Integration pipelineRuns", func() {
+				Expect(f.AsKubeDeveloper.IntegrationController.WaitForFinalizerToGetRemovedFromAllIntegrationPipelineRuns(testNamespace, applicationName, snapshot)).To(Succeed())
+			})
+
+			It("checks that when deleting integration test scenario pipelineRun, spaceRequest is deleted too", func() {
+				integrationPipelineRun, err = f.AsKubeAdmin.IntegrationController.GetIntegrationPipelineRun(integrationTestScenario.Name, snapshot.Name, testNamespace)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(f.AsKubeDeveloper.TektonController.DeletePipelineRun(integrationPipelineRun.Name, integrationPipelineRun.Namespace)).To(Succeed())
+
+				timeout = time.Second * 200
+				interval = time.Second * 5
+				Eventually(func() error {
+					spaceRequests, err := f.AsKubeAdmin.GitOpsController.GetSpaceRequests(testNamespace)
```

Comment:
I think we could shorten the code in the test by creating a reusable method `GetSpaceRequest` in `pkg/clients/gitops/space_requests.go` and wait until we get `k8sErrors.IsNotFound(err)` here, wdyt?

Reply:
done